src/eve/utils.py

```diff
@@ -486,22 +536,27 @@ class FrozenNamespace(types.SimpleNamespace, Generic[T]):
         >>> ns = FrozenNamespace(a=10, b="hello")
         >>> list(ns.items())
         [('a', 10), ('b', 'hello')]
+
+        >>> ns = FrozenNamespace(a=10, b="hello")
+        >>> hashed = hash(ns)
+        >>> assert isinstance(hashed, int)
+        >>> hashed == hash(ns) == ns.__cached_hash_value__
+        True
     """
 
-    def __setattr__(self, _name: str, _value: T) -> None:
-        raise TypeError(f"Trying to modify immutable '{self.__class__.__name__}' instance.")
+    __slots__ = "__cached_hash_value__"  # This slot is used to avoid polluting the namespace
```

Comment:
The slot is used to store the hash value without polluting the instance's `__dict__` used by `SimpleNamespace`. Not strictly needed but it's a cheap and efficient way to avoid reevaluation of the hash for the same items every time is called.